gui/src/model.rs

```diff
@@ -30,43 +30,52 @@ pub struct Settings {
     pub light_positions: Vec<Vec3>,
 }
 
-impl From<&Model> for Settings {
-    fn from(model: &Model) -> Self {
-        let finger_key_positions = model
-            .key_positions
-            .columns
-            .iter()
-            .flat_map(|column| column.iter().flat_map(mirrored_positions))
-            .collect();
-        let thumb_key_positions = model
-            .key_positions
-            .thumb_keys
-            .iter()
-            .flat_map(mirrored_positions)
-            .collect();
-        let interface_pcb_positions =
-            mirrored_positions(&model.keyboard.interface_pcb_position).to_vec();
-
-        let matrix_pcb_meshes = model
-            .matrix_pcb
-            .key_connectors
-            .iter()
-            .map(Into::into)
-            .chain(model.matrix_pcb.column_connectors.iter().map(Into::into))
-            .chain(once((&model.matrix_pcb.cluster_connector).into()))
-            .collect();
-        let fpc_pad_positions = mirrored_positions(&model.matrix_pcb.fpc_pad_position).to_vec();
-        let light_positions = light_positions_from_bounds(model.keyboard.right_half.bounds());
+#[derive(Clone, Default)]
+pub struct ThumbKeySettings {
+    pub key_size: config::KeySize,
```

Comment:
```suggestion
    /// The size of the thumb keys.
    pub key_size: config::KeySize,
```

Also please replace qualified path with use.